stader-labs/stader-node
# Integration Testing setup

### How to test this locally:

* `brew install kurtosis-tech/tap/kurtosis-cli` to install kurtosis
* ` kurtosis engine start` to start engine
* `cd testing && go test ./... -v` 

### What's in this PR:

[X] Eth1/eth2 run in docker container manage by `kurtosis_context` in our Go test code
- Eth1 by anvil container
- Eth2 by kurtosis

[X] Smart contracts deploy to eth1 (anvil)

[X] Start node in separate goroutine to start node

[X] Config file + data at `~/.stader_testing` folder
- ~/.stader_testing separate testing data for each suite

[X] Init wallet
[X] Http-test server

- http test server assert pre-sign message

[X] Verify pre-sign messages 
- Verify signature
- Verify decrypt / encrypt RSA
